lib/rucio/core/replica.py

```diff
@@ -816,11 +840,12 @@ def _list_replicas(dataset_clause, file_clause, state_clause, show_pfns, schemes
                         'pfns': {}, 'rses': defaultdict(list),
                         'states': {rse: str(state)}}
                 if rse:
-                    file['rses'][rse] = pfns
-                    for pfn in pfns:
-                        file['pfns'][pfn] = {'rse': rse,
-                                             'type': str(rse_type),
-                                             'volatile': volatile}
+                    file['rses'][rse] = list(set([tmp_pfn[1] for tmp_pfn in pfns]))  # extract properly the pfn from the (domain, pfn)
+                    for tmp_pfn in pfns:
+                        file['pfns'][tmp_pfn[1]] = {'rse': rse,
+                                                    'type': str(rse_type),
+                                                    'volatile': volatile,
+                                                    'domain': tmp_pfn[0]}  # extract properly the domain from the (domain, pfn)
```

Comment:
hard to know the possible values in tmp_pfn[0] from the review interface. I guess is 'wan', 'lan', 'all' ? 
One alternative is to have a dictionary like {'wan': True, 'lan': True }

Reply:
it's either wan or lan. all is just an option to the client to retrieve both. an extra dictionary would just make it even more confusing imho.

Reply:
a pfn can be on all domains: wan and lan no ?  What will be the tmp_pfn value if the pfn is only on lan, wan and both on wan and lan ?

Reply:
the domain is just a description of the access. if there's a both "lan" and "wan" access available for a pfn, but it's the same, then it's "wan". there's no need to separate it?

(unless we want to provide a usecase where we have something that is lan-only accessible, but not accessible via wan)

Reply:
that's the question :) Does the clients need to know that a pfn is accessible locally and then remotely ? e.g. If the clients first tries the lan replica and then fallbacks to a remote one ?

Reply:
those will be two different pfns in the list_replica reply ;-)

Reply:
pfn is the dictionary key thus unique now ?

Reply:
yes, of course. either you have a lan-accessible PFN, or a lan+wan-accessible PFN. you cannot have a wan-accessible PFN that is not lan-accessible. 

and depending if you request domain lan,wan,all you get back either 1 pfn (tagged with domain:lan, or tagged with domain:wan) or potentially 2 pfns (lan-only pfn: tagged with domain:lan; lan+wan pfn tagged with domain:wan).

default is still to reply with wan-accessible only.

Reply:
yes, a wan pfn can be accessible from the lan but it might not reflect the preferences and settings as expressed in Rucio (https for wan but not defined in lan): if the value 'all' should be there or not and if it brings more information.

Reply:
If sites have LAN preferences then they just need to set them. Those that do have the settings in the RSE configs already.